File: MATTHEW/leetcode/AIcoding/5.Qwen-3/javascript_solutions/858-mirror-reflection.js
var mirrorReflection = function(p, q) {
    let m = p, n = q;
    while (n) {
        let temp = n;
        n = m % n;
        m = temp;
    }
    let lcm = p * q / m;
    let a = lcm / p;
    let b = lcm / q;
    if (a % 2 === 0) return 2;
    if (b % 2 === 0) return 1;
    return 0;
};

File: leetcode/AIcoding/5.Qwen-3/rust_solutions/858-mirror-reflection.rs
struct Solution {}

impl Solution {
    pub fn mirror_reflection(p: i32, q: i32) -> i32 {
        let mut m = p;
        let mut n = q;

        while m % n != 0 {
            let temp = m % n;
            m = n;
            n = temp;
        }

        let lcm = p * q / n;

        let a = lcm / p;
        let b = lcm / q;

        if a % 2 == 0 {
            return 2;
        } else if b % 2 == 0 {
            return 1;
        } else {
            return 0;
        }
    }
}